tasks/IcomM510Task.cpp

```diff
@@ -0,0 +1,173 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "IcomM510Task.hpp"
+#include <nmea2000/PGNs.hpp>
+
+using namespace nmea2000;
+
+IcomM510Task::IcomM510Task(std::string const& name)
+    : IcomM510TaskBase(name)
+{
+}
+
+IcomM510Task::~IcomM510Task()
+{
+}
+
+bool IcomM510Task::configureHook()
+{
+    if (!IcomM510TaskBase::configureHook())
+        return false;
+    return true;
+}
+bool IcomM510Task::startHook()
+{
+    if (!IcomM510TaskBase::startHook())
+        return false;
+    return true;
+}
+void IcomM510Task::updateHook()
+{
+    IcomM510TaskBase::updateHook();
+    Message msg;
+    while (_msg_in.read(msg) == RTT::NewData) {
```

Comment:
As far as I understood, we won't need this anymore. But.

This was rather obviously a very good candidate to be copied in the nmea2000 library and used in both tasks, instead of this brutal copy/paste.